tachyon/math/finite_fields/binary_field.h

```diff
@@ -40,10 +46,14 @@ class BinaryField final : public FiniteField<BinaryField<_Config>> {
 
   constexpr BinaryField() = default;
   constexpr explicit BinaryField(Type value) : value_(value) {
-    DCHECK_LE(value_, GetMax());
+    if constexpr (kBits <= 8) {
+      DCHECK_LE(value_, GetMax());
+    }
```

Comment:
This is because there's no int type for `uint2_t` and `uint4_t`. For those binary fields whose modulus bit is greater than 8, they have a dedicated integer type, such as `uint16_t`.